shared/src/main/java/net/blay09/mods/waystones/recipe/WarpPlateRecipe.java

```diff
@@ -0,0 +1,121 @@
+package net.blay09.mods.waystones.recipe;
+
+import net.blay09.mods.waystones.block.ModBlocks;
+import net.blay09.mods.waystones.block.entity.WarpPlateBlockEntity;
+import net.minecraft.core.NonNullList;
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.*;
+import net.minecraft.world.level.Level;
+
+import java.util.stream.Stream;
+
+public class WarpPlateRecipe implements Recipe<WarpPlateBlockEntity> {
+    private final Ingredient innerIngredient;
+    private final Ingredient outerIngredient;
```

Comment:
instead of having just a single outer ingredient, it would be good if the outer parts would be treated like a shapeless recipe, i.e. taking 4 individual (but shapedless) ingredients, so that recipes like flint + 2 glowstone + 2 redstone would also be supported

Reply:
Supporting both "outer" and "outers" format adds a lot of unnecessary complexity. Since the benefit gained by having the single-ingredient "outer" variant is miniscule, especially considering how rarely this type of recipe will be defined, we should keep things simple and have just the one explicit variant of defining them.